spec/classes/candlepin_spec.rb

```diff
@@ -163,59 +155,71 @@
           end
-
-          it do
-            is_expected.to contain_cpdb_create('candlepin').
-              with_ssl_options("?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory")
-            is_expected.to contain_cpdb_update('candlepin').
-              with_ssl_options("?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory")
-          end
         end
 
         context 'with SSL CA defined' do
           let(:params) { super().merge(db_ssl: true, db_ssl_ca: '/etc/ssl/postgresql.pem') }
 
           it { is_expected.to compile.with_all_deps }
           it { is_expected.to contain_concat('/etc/candlepin/candlepin.conf') }
+          it { is_expected.not_to contain_cpdb_update('candlepin') }
           it do
             is_expected.to contain_concat_fragment('PostgreSQL Database Configuration').
               with_content(sensitive(/^jpa.config.hibernate.connection.url=jdbc:postgresql:\/\/psql\.example\.com:5432.*ssl=true/)).
               with_content(sensitive(/^org.quartz.dataSource.myDS.URL=jdbc:postgresql:\/\/psql\.example\.com:5432.*ssl=true/)).
               with_content(sensitive(/^jpa.config.hibernate.connection.url=.*sslrootcert=\/etc\/ssl\/postgresql.pem/)).
               with_content(sensitive(/^org.quartz.dataSource.myDS.URL=.*sslrootcert=\/etc\/ssl\/postgresql.pem/))
           end
+        end
 
-          it do
-            is_expected.to contain_cpdb_create('candlepin').
-              with_ssl_options("?ssl=true&sslrootcert=/etc/ssl/postgresql.pem")
-            is_expected.to contain_cpdb_update('candlepin').
-              with_ssl_options("?ssl=true&sslrootcert=/etc/ssl/postgresql.pem")
+        context 'with db_manage_on_startup Halt' do
+          let(:params) { super().merge(db_manage_on_startup: 'Halt') }
+
+          it 'migrates the database' do
+            is_expected.to contain_cpdb_update('candlepin')
+              .that_subscribes_to(['Package[candlepin]', 'Concat[/etc/candlepin/candlepin.conf]'])
+              .that_notifies('Service[tomcat]')
+          end
+        end
+
+        context 'with db_manage_on_startup Report' do
+          let(:params) { super().merge(db_manage_on_startup: 'Report') }
+
+          it 'migrates the database' do
+            is_expected.to contain_cpdb_update('candlepin')
+              .that_subscribes_to(['Package[candlepin]', 'Concat[/etc/candlepin/candlepin.conf]'])
+              .that_notifies('Service[tomcat]')
+          end
+        end
+
+        context 'with db_manage_on_startup None' do
+          let(:params) { super().merge(db_manage_on_startup: 'None') }
+
+          it 'migrates the database' do
+            is_expected.to contain_cpdb_update('candlepin')
+              .that_subscribes_to(['Package[candlepin]', 'Concat[/etc/candlepin/candlepin.conf]'])
+              .that_notifies('Service[tomcat]')
           end
```

Comment:
I've seen this be written as a loop.
```suggestion
        ['Halt', 'Report', 'None'].each do |value|
          context "with db_manage_on_startup #{value}" do
            let(:params) { super().merge(db_manage_on_startup: value) }

            it 'migrates the database' do
              is_expected.to contain_cpdb_update('candlepin')
                .that_subscribes_to(['Package[candlepin]', 'Concat[/etc/candlepin/candlepin.conf]'])
                .that_notifies('Service[tomcat]')
            end
          end
        end
```

Reply:
For now I am going to opt to keep it straight forward, as loops have been harder to understand or peel apart in tests sometimes.